src/main/java/microsoft/exchange/webservices/data/HttpClientWebRequest.java

```diff
@@ -57,11 +59,16 @@ public HttpClientWebRequest(CloseableHttpClient httpClient, HttpClientContext ht
    * Releases the connection by Closing.
    */
   @Override
-  public void close() {
-    if (null != httpPost) {
+  public void close() throws IOException {
+    // First check if we can close the response, by consuming the complete response
+    // This releases the connection but keeps it alive for future requests
+    // If that is not possible, we simply cleanup the whole connection
+    if (response != null && response.getEntity() != null) {
+      EntityUtils.consume(response.getEntity());
+    } else if (httpPost != null) {
       httpPost.releaseConnection();
-      //postMethod.abort();
     }
+
     httpPost = null;
```

Comment:
I don't think so. One of the invariables of this class seems to be something like:

`if (httpPost != null) then the response has not been closed yet`

So we have to set it to `null` to make sure we do not drop the connection if we call `close()` for a second time.
Maybe we also should set `response` to `null`, but I'm not entirely sure nothing in the library calls `getInputStream()` or `getErrorStream()` after `close()`, so I chose not to change that behaviour.